pyteal/ast/router.py

```diff
@@ -309,10 +314,18 @@ def to_cond_node(self, use_frame_pt: bool = False) -> CondNode:
 CondWithMethod.__module__ = "pyteal"
 
 
-@dataclass
 class ASTBuilder:
-    conditions_n_branches: list[CondNode] = field(default_factory=list)
-    methods_with_conds: list[CondWithMethod] = field(default_factory=list)
+    def __init__(self):
+        self.conditions_n_branches: list[CondNode] = []
+        self.methods_with_conds: list[CondWithMethod] = []
+        self._bare_cnbs: list[CondNode] = []
+
+    def _clean(self) -> None:
+        self._bare_cnbs = []
+        # self.conditions_n_branches = []
```

Comment:
experimental crud... this is a draft PR after all

Reply:
what is the purpose of this `_clean` function?

Reply:
Hmmm... this could be wrong...